src/MEVCapturingHook.sol

```diff
@@ -25,14 +30,38 @@ contract MEVCapturingHook is BaseHook {
             beforeRemoveLiquidity: false,
             afterAddLiquidity: false,
             afterRemoveLiquidity: false,
-            beforeSwap: false,
+            beforeSwap: true,
             afterSwap: false,
             beforeDonate: false,
             afterDonate: false,
-            beforeSwapReturnDelta: false,
+            beforeSwapReturnDelta: true,
             afterSwapReturnDelta: false,
             afterAddLiquidityReturnDelta: false,
             afterRemoveLiquidityReturnDelta: false
         });
     }
+
+    function _beforeSwap(address, PoolKey calldata key, IPoolManager.SwapParams calldata params, bytes calldata)
+        internal
+        override
+        returns (bytes4, BeforeSwapDelta, uint24)
+    {
+        // take a fee based on the priority fee
+        // and donate it to LP
+
+        uint256 BASE_AMOUNT = 1 wei; // ?? this is too low
+        uint256 priorityFee = tx.gasprice - block.basefee;
+        uint256 fee = priorityFee * BASE_AMOUNT;
+
+        // TODO: only charge fee in first swap of the block
+        // TODO: only charge dynamic fee if priority fee > some threshold
+
+        if (params.zeroForOne) {
+            manager.donate(key, fee, 0, "");
+        } else {
+            manager.donate(key, 0, fee, "");
+        }
+
+        return (BaseHook.beforeSwap.selector, toBeforeSwapDelta(int128(int256(fee)), 0), 0);
```

Comment:
we prob need to check the units as if token is WBTC, won't be the same as if it is a memecoin. 1 pt of fee can be a big diff later in this.

Reply:
this can now be set per pool by the owner of the contract
if not explicitly set it falls back to the default values